Packages/com.unity.inputsystem/InputSystem/Editor/Analytics/InputActionCodeAuthoringAnalytic.cs

```diff
@@ -0,0 +1,168 @@
+using UnityEngine.Serialization;
+
+namespace UnityEngine.InputSystem.Editor
+{
+#if UNITY_2023_2_OR_NEWER
+    [UnityEngine.Analytics.AnalyticInfo(eventName: kEventName, maxEventsPerHour: kMaxEventsPerHour,
+        maxNumberOfElements: kMaxNumberOfElements, vendorKey: UnityEngine.InputSystem.InputAnalytics.kVendorKey)]
+#endif // UNITY_2023_2_OR_NEWER
+    internal class InputActionCodeAuthoringAnalytic : UnityEngine.InputSystem.InputAnalytics.IInputAnalytic
+    {
+        public const string kEventName = "input_action_code_authoring";
+        public const int kMaxEventsPerHour = 100; // default: 1000
+        public const int kMaxNumberOfElements = 100; // default: 1000
+
+        /// <summary>
+        /// Enumeration type for code authoring APIs mapping to <see cref="InputActionSetupExtensions"/>.
+        /// </summary>
+        /// <remarks>
+        /// This enumeration type may be added to, but NEVER changed, since it would break older data.
+        /// </remarks>
+        public enum Api
+        {
+            AddBinding = 0,
+            AddCompositeBinding = 1,
+            ChangeBinding = 2,
+            ChangeCompositeBinding = 3,
+            Rename = 4,
+            AddControlScheme = 5,
+            RemoveControlScheme = 6,
+            ControlSchemeWithBindingGroup = 7,
+            ControlSchemeWithDevice = 8,
+            ControlSchemeWithRequiredDevice = 9,
+            ControlSchemeWithOptionalDevice = 10,
+            ControlSchemeOrWithRequiredDevice = 11,
+            ControlSchemeOrWithOptionalDevice = 12
+        }
+
+        private static readonly int[] m_Counters = new int[Enum.GetNames(typeof(Api)).Length];
+
+        /// <summary>
+        /// Registers a call to the associated API.
+        /// </summary>
+        /// <param name="api">Enumeration identifying the API.</param>
+        public static void Register(Api api)
+        {
+            if (suppress)
+                return;
+
+            // Note: Currently discards detailed information and only sets a boolean (aggregated) value.
+            ++m_Counters[(int)api];
+        }
+
+        /// <summary>
+        /// Suppresses the registration of analytics.
+        /// </summary>
+        /// <remarks>
+        /// May be used to temporarily suppress analytics to avoid false positives from internal usage.
+        /// </remarks>
+        public static bool suppress
+        {
+            get; set;
+        }
+
+        // Cache delegate
+        private static readonly Action<PlayModeStateChange> PlayModeChanged = OnPlayModeStateChange;
+
+        // Note: Internal visibility to simplify unit testing
+        internal static void OnPlayModeStateChange(PlayModeStateChange change)
+        {
+            Debug.Log("Play mode state change: " + change);
+            if (change == PlayModeStateChange.ExitingEditMode)
+            {
+                // Reset all counters when exiting edit mode
+                for (var i = 0; i < m_Counters.Length; ++i)
```

Comment:
Array.Clear(m_Counters, 0, m_Counters.Length)?

Reply:
Of course, thanks. Now updated.